dumper-integration-tests/redshift-tests/src/test/java/com/google/integration/SvvColumnsTest.java

```diff
@@ -0,0 +1,99 @@
+import static com.google.edwmigration.dumper.base.TestConstants.URL_DB;
+import static com.google.edwmigration.dumper.base.TestConstants.USERNAME_DB;
+import static com.google.edwmigration.dumper.jdbc.JdbcUtil.getDbColumnNames;
+import static com.google.edwmigration.dumper.jdbc.JdbcUtil.getStringNotNull;
+import static com.google.edwmigration.dumper.sql.SqlUtil.getSql;
+import static org.testng.Assert.assertEquals;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.LinkedHashMultiset;
+import com.opencsv.CSVReaderHeaderAware;
+import com.opencsv.CSVReaderHeaderAwareBuilder;
+import com.opencsv.exceptions.CsvException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Map;
+import org.testng.annotations.Test;
+
+public class SvvColumnsTest {
+
+  private static final String SQL_PATH = SQL_REQUESTS_BASE_PATH + "svv_columns.sql";
+  private static final String CSV_FILE_PATH = EXPORTED_FILES_BASE_PATH + "svv_columns.csv";
+
+  @Test
+  public void svvColumnsTest() throws SQLException, IOException, CsvException {
+    LinkedHashMultiset<String> dbMultiset = LinkedHashMultiset.create();
+    LinkedHashMultiset<String> csvMultiset = LinkedHashMultiset.create();
+    int dbColumnCount, csvColumnCount;
+    LinkedHashMultiset<String> dbColumnHeaders, csvColumnHeaders;
+
+    try (Connection connection = DriverManager.getConnection(URL_DB, USERNAME_DB, PASSWORD_DB);
+        PreparedStatement preparedStatement = connection.prepareStatement(getSql(SQL_PATH))) {
+      ResultSet rs = preparedStatement.executeQuery();
+
+      dbColumnCount = rs.getMetaData().getColumnCount();
+      dbColumnHeaders = LinkedHashMultiset.create(getDbColumnNames(rs.getMetaData()));
+
+      while (rs.next()) {
+        StringBuilder sb = new StringBuilder();
+
+        for (int i = 1; i <= dbColumnCount; i++) {
+          sb.append(getStringNotNull(rs, i));
+          if (i < dbColumnCount) {
+            sb.append(",");
+          }
+        }
+        dbMultiset.add(sb.toString());
+      }
+    }
+
+    try (FileReader fileReader = new FileReader(CSV_FILE_PATH);
+        CSVReaderHeaderAware reader =
+            new CSVReaderHeaderAwareBuilder(fileReader).withCSVParser(CSV_PARSER).build()) {
+      String[] line;
+      while ((line = reader.readNext()) != null) {
+        csvMultiset.add(Joiner.on(",").join(line));
+      }
+    }
+
+    try (FileReader fileReader = new FileReader(CSV_FILE_PATH);
+        CSVReaderHeaderAware reader =
+            new CSVReaderHeaderAwareBuilder(fileReader).withCSVParser(CSV_PARSER).build()) {
+      Map<String, String> lineMap = reader.readMap();
+      csvColumnCount = lineMap.keySet().size();
+      csvColumnHeaders = LinkedHashMultiset.create(lineMap.keySet());
+    }
```

Comment:
I don't like that implementation, but I think we just have to go further with this PR and improve in the future.

There is a problem with `reader.readMap();`. It has functionality the same as `reader.readNext()`, but it also gives you headers, although because it stores the data as a HashMap the sorting of data and headers are not natural. So, you can read the headers and the value in one iteration, everything will be shuffled.

Reply:
Resolving, will review once changes are submitted.